src/mbgl/style/sources/vector_source.cpp

```diff
@@ -18,78 +18,17 @@ VectorSource::VectorSource(std::string id,
                            variant<std::string, Tileset> urlOrTileset_,
                            std::optional<float> maxZoom_,
                            std::optional<float> minZoom_)
-    : Source(makeMutable<Impl>(std::move(id))),
-      urlOrTileset(std::move(urlOrTileset_)),
+    : TileSource(id, urlOrTileset_, util::tileSize_I, SourceType::Vector),
```

Comment:
I chose to have `tileSize` be a part of the base `TileSource` instead of making it specific to `RasterSource` since previously `VectorSource` used this value for tileSource [here](https://github.com/maplibre/maplibre-native/blob/1899cde3bd19dac0b88536c55d7a42b7507460ff/src/mbgl/style/sources/vector_source.cpp#L80). This also means the `getTileSize` function is available on `VectorSource`, which doesn't seem problematic. An argument can be made for specializing `tileSize` to `RasterSource` and its sub-classes, but that would entail a bigger change and less shared code.